### PR TITLE
Add testnet-tds support to testnet manager

### LINKED ISSUE #4629: Setup TdS testnet
For TdS, we'll want the following testnet setup:
1. Cluster endpoint of tds.solana.com (assuming we don't come up with a better name...)
1. Airdrops disabled (depends-on: #4540)
1. Genesis block provisioned with an initial set of accounts for all the participants (depends-on: #4539)
1. A small handful of Solana-run validators (3 including the bootstrap leader?).  CPU only for initial setup and testing but then we'll flip them to GPU when TdS starts.
1. A blockexplorer node
1. A bench-tps client that for initial testing generates some baseline load (say ~1000 TPS)
1. Manual buildkite setup to manage the testnet, like what we did for the "demo testnet".  The ledger should not automatically reset every 24 hours, the cluster just runs until it's dead 💀 
1. Use the "auto" configuration for hashes-per-tick
1. Setup the genesis block to mint 500M SOL to ensure a high-rate of inflation during the stage (`--lamports` set to 500M*2^34)

### PR DESCRIPTION
**Pull request by danpaul000**
#### Problem

Need to set up an on demand testnet for Tour de Sol

#### Summary of Changes

Add testnet-tds support to testnet-manager and build on-demand pipelines in Buildkite

Fixes #4629

### CODE REVIEW

**Comment by danpaul000** [comment]
This PR covers almost all of the requests in #4629 except that we can't explicitly limit to 1000 TPS.  I am seeing a repeated failure to boot in the buildkite job that I have yet to crack: https://buildkite.com/solana-labs/testnet-tds-enable/builds/27#5ba44650-f1fa-41f9-8383-28219bc812e6

File: net/net.sh [hunk 1]
```diff
@@ -96,6 +98,7 @@ failOnValidatorBootupFailure=true
 genesisOptions=
 numFullnodesRequested=
 externalPrimordialAccountsFile=
+remoteExternalPrimordialAccountsFile=/tmp/external-primodial-accounts.yml
```
Comment (mvines) [inline net/net.sh]: Who populates /tmp/external-primodial-accounts.yml?   Perhaps `remoteExternalPrimordialAccountsFile` should be passed in as an argument to `net.sh`?

File: ci/testnet-manager.sh [hunk 2]
```diff
@@ -444,6 +459,26 @@ deploy() {
...
+      NO_LEDGER_VERIFY=1 \
+      NO_VALIDATOR_SANITY=1 \
+        ci/testnet-deploy.sh -p tds-solana-com -C gce ${GCE_ZONE_ARGS[0]} \
+          -t "$CHANNEL_OR_TAG" -n "$GCE_NODE_COUNT" -c 1 -P -u \
+          -a tds-solana-com --hashes-per-tick auto \
+          ${skipCreate:+-e} \
+          ${skipStart:+-s} \
+          ${maybeStop:+-S} \
+          ${maybeDelete:+-D} \
+          -k 1000 -K /tmp/stakes.yml \
```
Comment (mvines) [inline ci/testnet-manager.sh]: Where does `/tmp/stakes.yml` come from?

File: ci/testnet-deploy.sh [hunk 3]
```diff
@@ -64,10 +67,12 @@ Deploys a CD testnet
    -f                   - Discard validator nodes that didn't bootup successfully
    -w                   - Skip time-consuming "bells and whistles" that are
                           unnecessary for a high-node count demo testnet
-
+   -k                   - Amount to stake internal nodes.  If set, airdrops are disabled.
```
Comment (mvines) [inline ci/testnet-deploy.sh]: Let's start using long opts for all new arguments  please.  The short args are basically unreadable (`-k`, `-K`) given that we've used almost every letter in the alphabet now

File: ci/testnet-manager.sh (hunk 2 above)
Comment (mvines) [inline ci/testnet-manager.sh]: `-k 1000` may be too small, that's enough for ~1 minute of voting.   I'd add many more zeros, this may be the boot issue

**Comment by mvines** [comment]
> I am seeing a repeated failure to boot in the buildkite job that I have yet to crack: https://buildkite.com/solana-labs/testnet-tds-enable/builds/27#5ba44650-f1fa-41f9-8383-28219bc812e6

It looks like the bootstrap leader starts up, but then all the other validators fail at https://github.com/solana-labs/solana/blob/f9a22546886016819ebc7e2613932dd2fb63fe2c/multinode-demo/fullnode.sh#L99-L101 with `Error: TransactionError(InstructionError(0, CustomError(1)))`.  I don't see anything in the bootstrap leader's log with more details.  The low lamports per node (`-k 1000`) looks suspicious.  Can you repro this locally with just the `net/net.sh` args?

File: net/net.sh (hunk 1 above)
Comment (danpaul000) [reply net/net.sh]: The file is rsync-ed over from the calling machine (in this case, the CI buildkite VM), then its contents are cat'ed out to a known file that contains the internal and external node staking.  It's just a temp file to get it onto the correct machine, so no need to have a dynamic path for this one.

File: ci/testnet-manager.sh (hunk 2 above)
Comment (danpaul000) [reply ci/testnet-manager.sh]: Currently it's a file sitting on m-s-c VM with some a dummy pubkey: stake data I created.  This path can point to anything that contains our external stake holders.  I think a cleaner option might be to wget this file from some managed location onto the machine running testnet-manager.sh, then pass it down from there.  I was planning to add that feature to a subsequent PR unless you think it is important to get in from the beginning here.

Comment (danpaul000) [reply ci/testnet-manager.sh]: That value was an arbitrary guess.  Is it worth pulling that data up to an env var (along with the /tmp/stakes.yml filepath?) so we can set it from buildkite job?

File: ci/testnet-deploy.sh (hunk 3 above)
Comment (danpaul000) [reply ci/testnet-deploy.sh]: Will do

File: ci/testnet-manager.sh (hunk 2 above)
Comment (mvines) [reply ci/testnet-manager.sh]: sg, follow-up PR totally fine by me

**Comment by danpaul000** [comment]
Comments have been addressed.  Block explorer seems a little flaky, but can be taken up in a subsequent PR.  Any other feedback here? @mvines @pgarg66

**Pull request merged**